web/src/app/converters/proto-model-converter.ts

```diff
@@ -99,28 +99,13 @@ export function jobToDocument(job: Job): DocumentData {
       index: job.index,
       name: job.name,
       style: new Pb.Style({color: job.color}),
-      tasks: job.tasks
-        ?.map(task => {
-          return new Pb.Task({
-            ...taskTypeToPartialMessage(task),
-            id: task.id,
-            index: task.index,
-            prompt: task.label,
-            required: task.required,
-            level: task.addLoiTask
-              ? Pb.Task.DataCollectionLevel.LOI_DATA
-              : Pb.Task.DataCollectionLevel.LOI_METADATA,
-            conditions: taskConditionToPartialMessage(task),
-          });
-        })
-        .toList()
-        .toArray(),
+      tasks: tasksToDocument(job.tasks?.toList() || List([])),
     })
   );
 }
 
 /**
- * Creates a partial rapresentation of a Task.
+ * Returns the proto representation of a partial Task model object.
```

Comment:
Why do we have so many separate methods to handle "partial" model objects? Aren't partial objects just regular objects with some values just set to `null`, or in the case of protos, being unset?